site/app/controllers/banner/BannerController.php

```diff
@@ -0,0 +1,195 @@
+     * @see BannerView::showBanner
+     */
+    public function viewBanner(): WebResponse {
+        $bannerImages = $this->core->getSubmittyEntityManager()->getRepository(BannerImage::class) ->findall();
+        return new WebResponse(BannerView::class, 'showBanner', $bannerImages);
+        //EVEN WHEN I REPLACE BannerView::class with 'app\views\banner\BannerView', webresponse still treats the parameter as 'app\\controllers\\Banner\\BannerView'
+    }
+
+
+
+    /**
+     * @Route("/banner/upload", methods={"POST"})
+     */
+    public function ajaxUploadBannerFiles(): JsonResponse {
+        $upload_path = FileUtils::joinPaths($this->core->getConfig()->getSubmittyPath(), "banner_images");
+        if (isset($_POST['release_time'])) {
+            $release_date = DateUtils::parseDateTime($_POST['release_time'], $this->core->getDateTimeNow()->getTimezone());
+        }
+        else {
+            return JsonResponse::getErrorResponse("No release date.");
+        }
+
+
+        if (isset($_POST['close_time'])) {
+            $close_date = DateUtils::parseDateTime($_POST['close_time'], $this->core->getDateTimeNow()->getTimezone());
+        }
+        else {
+            return JsonResponse::getErrorResponse("No release date.");
+        }
+        if (!isset($_FILES["files1"]) || empty($_FILES["files1"])) {
+            return JsonResponse::getErrorResponse("No files were submitted.");
+        }
+
+        $uploaded_files = $_FILES["files1"];
+        $count_item = count($uploaded_files["name"]);
+        $extra_name = $_POST['extra_name'];
+
+        if ($extra_name == "..") {
+            return JsonResponse::getErrorResponse("invalid name");
+        }
+
+        if (empty($extra_name) || strpos($extra_name, 'http') !== false) {
+            if ($count_item !== 1) {
+                return JsonResponse::getErrorResponse("You can only have one banner submitted.");
+            }
+        }
+        else {
+            if ($count_item > 2) {
+                return JsonResponse::getErrorResponse("Can't have more than two banners submitted.");
+            }
+        }
+
+        $specificPath = $close_date->format("Y");
+        $full_path = FileUtils::joinPaths($upload_path, $specificPath);
+
+
+
+        if (!is_dir($full_path)) {
+            // Create a new folder for the current month
+            if (!mkdir($full_path, 0755, true)) {
+                return JsonResponse::getErrorResponse("Failed to create a new folder for the current year.");
+            }
+        }
+
+        for ($j = 0; $j < $count_item; $j++) {
+            if ($uploaded_files['name'][$j] == "..") {
+                return JsonResponse::getErrorResponse("invalid name");
+            }
+            
+            // for some reason why I try to simply use a condition to compare two strings, I always get false?!? So I have to loop through each character now
+            $all_match = true;
+            for ($i = 0; $i < strlen($uploaded_files['name'][$j]); $i++) {
```

Comment:
PHP provides string comparison functions, for instance, strcmp($str1, $str2) will return 0 if equal, positive and negative depending on the which string is greater. There are also many variations to it, (strcasecmp etc).